rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java

```diff
@@ -219,11 +219,11 @@ private boolean validateLogsBloom(BlockHeader header, BlockResult result) {
 
     @VisibleForTesting
     public BlockResult execute(Block block, BlockHeader parent, boolean discardInvalidTxs) {
-        return execute(block, parent, discardInvalidTxs, false);
+        return execute(block, parent, discardInvalidTxs, false, true);
     }
 
-    public BlockResult execute(Block block, BlockHeader parent, boolean discardInvalidTxs, boolean ignoreReadyToExecute) {
-        return executeInternal(null, 0, block, parent, discardInvalidTxs, ignoreReadyToExecute);
+    public BlockResult execute(Block block, BlockHeader parent, boolean discardInvalidTxs, boolean ignoreReadyToExecute, boolean saveState) {
```

Comment:
WDYT about splitting this method into two, one called `executeSavingState` and another one `executeSkippingStateSave` (or sth like that) instead of passing a boolean? I think it is more clear.

Reply:
I like @asoto-iov approach mentioned [here](https://github.com/rsksmart/rskj/pull/1885#discussion_r976781685), I think it is the best solution.

The reason I suggested this intermediate approach is to avoid a refactor, as it would be changing the same places you have already changed. Up to you.